benchmarks/transport_montecarlo_numba_formal_integral_p.py

```diff
@@ -102,75 +124,75 @@ def time_populate_z_photosphere(self, p, test_data):
-        size = len(self.formal_integral_geometry(test_data).r_outer)
         r_inner = self.formal_integral_geometry(test_data).r_inner
         self.formal_integral_geometry(test_data).r_outer
 
         p = r_inner[0] * p
         oz = np.zeros_like(r_inner)
         oshell_id = np.zeros_like(oz, dtype=np.int64)
 
-        n = func(
+        func(
             self.formal_integral_geometry(test_data),
             self.formal_integral_geometry(test_data),
             p,
             oz,
             oshell_id,
         )
-        assert n == size
 
     @skip_benchmark
     @parameterize({"p": [1e-5, 0.5, 0.99, 1], "Test data": TESTDATA})
-    def time_populate_z_shells(self, p, test_data):
-        formal_integral.FormalIntegrator(
-            self.formal_integral_geometry(test_data), None, None
-        )
+    def time_populate_z_shells(self, p, test_data) -> None:
         func = formal_integral.populate_z
 
         size = len(self.formal_integral_geometry(test_data).r_inner)
         r_inner = self.formal_integral_geometry(test_data).r_inner
         r_outer = self.formal_integral_geometry(test_data).r_outer
 
         p = r_inner[0] + (r_outer[-1] - r_inner[0]) * p
-        idx = np.searchsorted(r_outer, p, side="right")
 
         oz = np.zeros(size * 2)
         oshell_id = np.zeros_like(oz, dtype=np.int64)
 
-        offset = size - idx
-
-        expected_n = (offset) * 2
-        expected_oz = np.zeros_like(oz)
-        expected_oshell_id = np.zeros_like(oshell_id)
-
-        # Calculated way to determine which shells get hit
-        expected_oshell_id[:expected_n] = (
-            np.abs(np.arange(0.5, expected_n, 1) - offset) - 0.5 + idx
-        )
-
-        expected_oz[0:offset] = 1 + self.calculate_z(
-            r_outer[np.arange(size, idx, -1) - 1], p
-        )
-        expected_oz[offset:expected_n] = 1 - self.calculate_z(
-            r_outer[np.arange(idx, size, 1)], p
-        )
-
-        n = func(
+        func(
             self.formal_integral_geometry(test_data),
             self.formal_integral_geometry(test_data),
             p,
             oz,
             oshell_id,
         )
 
-        assert n == expected_n
-
-    @parameterize({"N": [100, 1000, 10000]})
-    def time_calculate_p_values(self, n):
+    @parameterize(
+        {
+            "Parameters": [
+                {
+                    "N": [100, 1000, 10000]
+                }
+            ]
+        }
+    )
+    def time_calculate_p_values(self, Parameters):
         r = 1.0
-        func = formal_integral.calculate_p_values
+        formal_integral.calculate_p_values(r, Parameters["N"])
 
-        expected = r / (n - 1) * np.arange(0, n, dtype=np.float64)
-        np.zeros_like(expected, dtype=np.float64)
+    @parameterize({
+            "parameters": {
+                "nu": [1e13, 5e14, 1e15],
+                "temperature": [300, 1000, 5800]
+            }
+        }
+    )
+    def time_intensity_black_body(self, nu, temperature):
```

Comment:
Why is the function name same as line 55?

Reply:
Sorry, that was a mistake. Updated the file!